manticore/platforms/linux.py

```diff
@@ -2728,7 +2728,10 @@ def syscall(self):
             else:
                 raise EnvironmentError(f"Bad syscall index, {index}")
 
-        return self._syscall_abi.invoke(implementation)
+        self._syscall_abi._cpu._publish("will_invoke_syscall", index)
```

Comment:
I think this might need to be moved higher up somewhere before the exception handling in case base manticore doesn't have the implementation but someone still wants to hook on it.